### PR TITLE
Replace "go get" to install tools with "go install"

### PR DESCRIPTION
**Pull request by ryysud**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

- Tools installation
- Unit tests in CI/CD

**Description of change**

To fix https://github.com/spiffe/spire/issues/2222, I considered removing the temporary directory for the tools. As a result, I decide to continue using the directory. Because it is difficult to check the version of tools and the Go version used to build tools that are installed with `go install` in $GOPATH* or $HOME*. The directory name contains both versions.

*Reference:

> Executables are installed in the directory named by the GOBIN environment variable, which defaults to $GOPATH/bin or $HOME/go/bin if the GOPATH environment variable is not set. Executables in $GOROOT are installed in $GOROOT/bin or $GOTOOLDIR instead of $GOBIN.

https://pkg.go.dev/cmd/go#hdr-Compile_and_install_packages_and_dependencies

The final changes are as follows:

- [Do not refer to go.mod in the go install version specification](https://github.com/spiffe/spire/commit/c6bd6a7a12615a54420aac9df3cca79a40e88971)
- [Replace "go get" to install tools with "go install"](https://github.com/spiffe/spire/commit/60d15b2cb7e7385d5dd5155bf01a2c4a729344d8)

**Which issue this PR fixes**

A part of https://github.com/spiffe/spire/issues/2222.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -141,7 +141,7 @@ protoc_gen_go_grpc_base_dir := $(build_dir)/protoc-gen-go-grpc
 protoc_gen_go_grpc_dir := $(protoc_gen_go_grpc_base_dir)/$(protoc_gen_go_grpc_version)-go$(go_version)
 protoc_gen_go_grpc_bin := $(protoc_gen_go_grpc_dir)/protoc-gen-go-grpc
 
-protoc_gen_go_spire_version := $(shell grep github.com/spiffe/spire-plugin-sdk go.mod | awk '{print $$2}')
+protoc_gen_go_spire_version := v1.2.1
```
Comment (MarcosDY) [inline Makefile:144]: can you revert this changes?

**Comment by azdagron** [comment]
Thank you for the changes to use `go install` in those coverage scrips, @ryysud! The removal of the temporary directory referenced by #2222 was already done in #2538. Looks like I forgot to set that checkbox!

@MarcosDY is correct though, we want to pick the version of those tools out of go.mod so that our tooling is aligned with the dependency version that SPIRE uses.

**Comment by ryysud** [comment]
Thank for your review, @MarcosDY and @azdagron !
I reverted https://github.com/spiffe/spire/commit/c6bd6a7a12615a54420aac9df3cca79a40e88971 and renamed pull-req title.

**Review by azdagron — APPROVED** [review]
Excellent! Thanks, @ryysud !

**Review by azdagron — APPROVED** [review]
